DataFormats/L1TMuonPhase2/interface/Constants.h

```diff
@@ -62,8 +62,9 @@ namespace Phase2L1GMT {
   const int BITSGTETA = 14;
   const int BITSGTZ0 = 10;
   const int BITSGTD0 = 10;
-  const int BITSGTQUAL = 8;
-  const int BITSGTISO = 4;
+  const int BITSGTQUAL = 6;
```

Comment:
Seeing that you reduced the bits for the interface even further, can we assume that the quality bitwidth has been fixed with this now? If I can recall internally your quality was 9 bits but wrapped into 8 on the interface, hence we saw mismatches when comparing the CMSSW hardware accurate emulator with the custom emulator from the menu team. Did you now split the old 9 bit quality into a quality score and quality flags field?

Reply:
The tkMuon quality used to store the matching quality + pt penalty, which used more than 8 bits. Instead of storing the matching quality, we have moved to store the tkmuon quality bits with various[ quality cuts ](https://github.com/zhenbinwu/cmssw/blob/GMT_24ESR/L1Trigger/Phase2L1GMT/src/TPSAlgorithm.cc#L86). These qualities are open for suggestions. We will hear feedback from menu team to further fine tune them. Moving to quality bits, we reduced the bit width from 8 to 6, which I think should be sufficient. Quality score is quality bits | (charge << 6). This is already done in [here](https://github.com/zhenbinwu/cmssw/blob/GMT_24ESR/DataFormats/L1TMuonPhase2/interface/TrackerMuon.h#L58). For backward capabilities, I still keep the charge and quality(). 

Since we will send isolation sum instead of isolation bit, I increased the isolation bit width from 4 to 6, with the LSB of 0.25GeV. The changes only for quality + isolation, so that the rest of the interface don't need changes. But I will update the interface doc.

Reply:
![image](https://github.com/cms-l1t-offline/cmssw/assets/8050292/b22f32a3-c1bb-4482-956d-ffde6057fbce)
I think there might be an oversight shouldn't it now rather be:

0 q
6-1 Q
12-7 Isolation
16-13 \beta

Reply:
Sorry, my oversight. Thanks for pointing it out. Updated.